src/braket/pennylane_plugin/ahs_device.py

```diff
@@ -0,0 +1,356 @@
+from functools import partial
+import numpy as np
+
+from braket.aws import AwsDevice
+from braket.devices import LocalSimulator
+from braket.ahs.atom_arrangement import AtomArrangement
+from braket.ahs.analog_hamiltonian_simulation import AnalogHamiltonianSimulation
+from braket.ahs.driving_field import DrivingField
+from braket.timings.time_series import TimeSeries
+
+from pennylane import QubitDevice
+from pennylane._version import __version__
+from pennylane.pulse.rydberg_hamiltonian import RydbergHamiltonian, RydbergPulse
+
+
+class BraketAhsDevice(QubitDevice):
+    """Abstract Amazon Braket device for analogue hamiltonian simulation with PennyLane.
+
+    Args:
+        wires (int or Iterable[Number, str]]): Number of subsystems represented by the device,
+            or iterable that contains unique labels for the subsystems as numbers
+            (i.e., ``[-1, 0, 2]``) or strings (``['ancilla', 'q1', 'q2']``).
+        shots (int): Number of executions to run to aquire measurements. Defaults to 100.
+    """
+
+    name = "Braket AHS PennyLane plugin"
+    pennylane_requires = ">=0.29.0"
+    version = __version__
+    author = "Xanadu Inc."
+
+    operations = {"ParametrizedEvolution"}
+
+    def __init__(
+            self,
+            wires,
+            device,
+            *,
+            shots=100):
+
+        if not shots:
+            raise RuntimeError(f"This device requires shots. Recieved shots={shots}")
```

Comment:
```suggestion
            raise RuntimeError(f"This device requires shots. Received shots={shots}")
```
Also the local device requires finite shots?

Reply:
Yes it does! It simulates the behaviour of the hardware.